Add using operator to primitive sequence

hagmas opened the pull request:
Hello, 

This PR is for adding `using` operator to `PrimitiveSequence`.

kzaher commented on RxSwift/Traits/PrimitiveSequence.swift:
```diff
@@ -579,6 +579,22 @@ extension PrimitiveSequence {
+     - parameter primitiveSequenceFactory: Factory function to obtain an observable sequence that depends on the obtained resource.
+     - returns: An observable sequence whose lifetime controls the lifetime of the dependent resource object.
+     */
+    public static func using<R: Disposable>(_ resourceFactory: @escaping () throws -> R, primitiveSequenceFactory: @escaping (R) throws -> PrimitiveSequence<Trait, Element>)
```
Let's rename `R: Disposable` -> `Resource: Disposable`. I understand that you've named it `R` because of consistency, but we want to improve the naming.

kzaher commented on RxSwift/Traits/PrimitiveSequence.swift:
```diff
@@ -579,6 +579,22 @@ extension PrimitiveSequence {
+     */
+    public static func using<R: Disposable>(_ resourceFactory: @escaping () throws -> R, primitiveSequenceFactory: @escaping (R) throws -> PrimitiveSequence<Trait, Element>)
+        -> PrimitiveSequence<Trait, Element> {
+            return PrimitiveSequence(raw: Observable.using(resourceFactory, observableFactory: { (r: R) throws -> Observable<E> in
```
Let's rename the parameter `r` -> `resource`.

kzaher commented on Tests/RxSwiftTests/PrimitiveSequenceTest.swift:
```diff
@@ -671,6 +671,52 @@ extension PrimitiveSequenceTest {
+        var createInvoked = 0
+        
+        var disposable: MockDisposable!
+        var xs:TestableObservable<Int>!
```
Can you please fix the whitespace to be consistent with the rest of the project.

hagmas commented:
Hello @kzaher,

Thank you vm for reviewing! I fixed some points as you commented. FWIW, I renamed R to ResourceType instead of Resource just to be consistent with 69#Using.swift. In addition, I did the same change for Using.swift as well (Let me know if this change is unnecessary. I will revert it.).

Thanks!

kzaher commented:
Hi @hagmas ,

I think we would prefer `Resource` over `ResourceType`.

In some places we had to use `*Type` naming because of name clashing between associated types in protocol and generic parameters.

hagmas commented:
Hi @kzaher ,

I understood. I renamed ResourceType to Resource. Could you please review again?

Thank you
